### PR TITLE
stop inserting 1.0 on zero block diagonal

### PR DESCRIPTION
**Pull request by msandre**
this fixes the problem with EigensolverStrategy::ApplyDirichletCondition() explained in #279.

### CODE REVIEW

File: applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp [hunk 1]
```diff
@@ -562,8 +562,8 @@ class EigensolverStrategy
                     empty = false;
                     break;
                 }
-            if(empty == true)
```
Comment (RiccardoRossi) [inline applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp]: @KratosMultiphysics/dem i would suggest you to comment out the whole loop on lines 553-567

**Review by RiccardoRossi — APPROVED** [review]
this commit makes sense to me (aside for the minor comment i made)

however i shall say that i did not test it...

File: applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp (hunk 1 above)
Comment (maceligueta) [reply applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp]: @KratosMultiphysics/dem ?

**Comment by qaumann** [comment]
I tested it and the eigensolver now gives the correct eigenfrequencies and mode shapes using the ResidualBasedBlockBuilderAndSolver

**Pull request merged**